workspace-service/src/graphql/folders.rs

```diff
@@ -96,8 +96,21 @@ impl FoldersQuery {
     ) -> FieldResult<Vec<Folder>> {
         let pool = context.data()?;
         let workspace = Uuid::parse_str(&workspace)?;
+        let requesting_user = context.data()?;
+        let event_client: &EventClient = context.data()?;
         let folders = db::FolderRepo::find_by_workspace(workspace, pool).await?;
-        Ok(folders.into_iter().map(Into::into).collect())
+        let first_folder = folders.first().unwrap();
+        let user_role =
+            get_workspace_membership(first_folder.workspace, requesting_user, pool, event_client)
+                .await?;
+        Ok(folders
+            .into_iter()
+            .map(Into::into)
+            .filter(|folder: &Folder| {
+                !(folder.role_required == RoleRequired::WorkspaceMember
+                    && user_role == WorkspaceMembership::NonMember)
```

Comment:
I just realized. I think this does potentially exclude platform admins. We should probably do something like:

```suggestion
                !(folder.role_required == RoleRequired::WorkspaceMember
                    && user_role == WorkspaceMembership::NonMember
                    && !user_is_platform_admin)
```

Reply:
Think this should be fixed by https://github.com/FutureNHS/futurenhs-platform/pull/504, which will declare any `platform_admin` as an Admin for any workspace.

Reply:
Ah brilliant.